tests/setns-report-ns-id.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Check appending auxstr of setns(2) when --namespace=new is given
+ *
+ * Copyright (c) 2024 The strace developers.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include "tests.h"
+#include "scno.h"
+#include "xmalloc.h"
+
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/socket.h>
+
+typedef struct msg {
+	enum {
+		MSG_OK = 0,
+		MSG_ERR_CLOSE,
+		MSG_ERR_UNSHARE,
+	} type;
+	int err;
+} msg_t;
+
+int
+main(void)
+{
+	int sv[2];
+	if (socketpair(AF_UNIX, SOCK_STREAM, 0, sv) < 0)
+		perror_msg_and_skip("socketpair");
+
+	pid_t pid = fork();
+	if (pid == 0) {
+		msg_t msg;
+		char c;
+
+		if (syscall(__NR_unshare, 0x10000000) < 0) {
```

Comment:
Wouldn't the test become simpler if it was the parent process that does unshare(CLONE_NEWUSER) instead of the child process?

Reply:
(WORK IN PROGRESS)

Reply:
fixed locally. I wonder why I designed the test case so complicated.

Reply:
Done.